vllm/executor/openvino_executor.py

```diff
@@ -0,0 +1,539 @@
+
+        self.head_size = model_config.get_head_size()
+        self.num_layers = model_config.get_num_layers(parallel_config)
+        self.num_heads = model_config.get_num_kv_heads(parallel_config)
+
+        self.block_size = cache_config.block_size
+        self.num_gpu_blocks = cache_config.num_gpu_blocks
+        self.num_cpu_blocks = cache_config.num_cpu_blocks
+
+        self.cache_dtype = OpenVINOCacheEngine.get_cache_dtype(
+            self.cache_config.cache_dtype,
+            self.model_config,
+            self.device_config)
+
+        # Initialize the cache.
+        self.gpu_cache = self.allocate_gpu_cache()
+        self.cpu_cache = self.allocate_cpu_cache()
+
+    def get_key_block_shape(self) -> Tuple[int, int, int, int]:
+        # TODO: ilavreno: we can insert per-device KV cache shapes configurations
+        # which are more optimal for plugins representations
+        element_size = self.cache_dtype.get_size()
+        x = 16 // element_size
+        return (
+            self.num_heads,
+            self.head_size // x,
+            self.block_size,
+            x,
+        )
+
+    def get_value_block_shape(self) -> Tuple[int, int, int]:
+        # TODO: ilavreno: we can insert per-device KV cache shapes configurations
+        # which are more optimal for plugins representations
+        return (
+            self.num_heads,
+            self.head_size,
+            self.block_size,
+        )
+
+    def allocate_gpu_cache(self) -> List[OpenVINOKVCache]:
+        if self.num_gpu_blocks == 0:
+            return None
+        gpu_cache: List[OpenVINOKVCache] = []
+        key_block_shape = (self.num_gpu_blocks, *self.get_key_block_shape())
+        value_block_shape = (self.num_gpu_blocks, *self.get_value_block_shape())
+        remote_context = ov.Core().get_default_context("GPU")
+        for _ in range(self.num_layers):
+            key_blocks = remote_context.create_tensor(self.cache_dtype, key_block_shape)
+            value_blocks = remote_context.create_tensor(self.cache_dtype, value_block_shape)
+            gpu_cache.append((key_blocks, value_blocks))
+        return gpu_cache
+
+    def allocate_cpu_cache(self) -> List[OpenVINOKVCache]:
+        cpu_cache: List[OpenVINOKVCache] = []
+        key_block_shape = (self.num_cpu_blocks, *self.get_key_block_shape())
+        value_block_shape = (self.num_cpu_blocks, *self.get_value_block_shape())
+        for _ in range(self.num_layers):
+            key_blocks = ov.Tensor(self.cache_dtype, key_block_shape)
+            value_blocks = ov.Tensor(self.cache_dtype, value_block_shape)
+            cpu_cache.append((key_blocks, value_blocks))
+        return cpu_cache
+
+    def _swap(
+        self,
+        src: List[OpenVINOKVCache],
+        dst: List[OpenVINOKVCache],
+        src_to_dst: Dict[int, int],
+    ) -> None:
+        # TODO: ilavreno: implement cache sync via OpenVINO RemoteContext API host <-> device
+        assert False
```

Comment:
@vladimir-paramuzov
Here we should insert code which copies data from host to device and back